packages/dds/tree/src/simple-tree/api/treeNodeApi.ts

```diff
@@ -129,6 +129,12 @@ export interface TreeNodeApi {
 	 * The same node's identifier may, for example, be different across multiple sessions for the same client and document, or different across two clients in the same session.
 	 */
 	shortId(node: TreeNode): number | string | undefined;
+
+	/**
+	 * Gets the metadata for the field associated with the provided key under the provided node.
+	 * Will return `undefined` if no metadata was provided for the field.
+	 */
+	fieldMetadata(node: TreeNode, key: string | number): unknown | undefined;
```

Comment:
I don't think you are providing a way to associate metadata with indexes of array nodes so this seems odd.

Also it seems odd this works on nodes not schema, so if you want top look up the metadata data you need an instance of the node. I think it would make more sense to just use Tree.schema(node) then lookup them metadata from that. Having the APi like this imples different node instances could have different metadata, which is not the case.